src/landing-page/landing-page.js

```diff
@@ -0,0 +1,293 @@
+/**
+ * WordPress dependencies
+ */
+import { sprintf, __ } from '@wordpress/i18n';
+import { useState, createInterpolateElement } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import {
+	Button,
+	ExternalLink,
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalVStack as VStack,
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalHStack as HStack,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { fetchThemeStyleData, downloadExportedTheme } from '../resolvers';
+import { downloadFile } from '../utils';
+import { CreateThemeModal } from './create-modal';
+
+export default function LandingPage() {
+	const [ themeStyleData, setThemeStyleData ] = useState( '' );
+	const [ createModalType, setCreateModalType ] = useState( false );
+
+	useSelect( async () => {
+		setThemeStyleData( await fetchThemeStyleData() );
+	}, [] );
```

Comment:
Referencing @t-hamano's comment, I've tried this and it works fine:

```suggestion
	const [ themeName, setThemeName ] = useState( '' );
	const [ createModalType, setCreateModalType ] = useState( false );

	const currentTheme = useSelect( ( select ) =>
		select( coreStore ).getCurrentTheme()
	);

	useEffect( () => {
		const name = currentTheme?.name?.raw || '';
		setThemeName( name );
	}, [ currentTheme ] );
```

Replacing references to `themeStyleData?.name` with `themeName`, and of course importing `useEffect` and the core data store:

```js
import {
	useState,
	createInterpolateElement,
	useEffect,
} from '@wordpress/element';
import { store as coreStore } from '@wordpress/core-data';
```

So I agree that we probably don't need the endpoint after all, at least not for the theme name.

Reply:
The core API should work outside of the editor.

However, if we implement it like this, it won't work because the dependencies don't include `wp-core-data`.

```javascript
const themeName = useSelect( ( select ) =>
	select( 'core' ).getCurrentTheme()
)?.name?.raw;
```

If we want to use the core API outside of the editor, we need to explicitly import the core library.

```javascript
import { store as coreStore } from '@wordpress/core-data';

const themeName = useSelect( ( select ) =>
	select( coreStore ).getCurrentTheme()
)?.name?.raw;
```

Additionally, the theme name is only used in the button text, so there should be no need to manage it as a state.

```javascript
export default function LandingPage() {
	// ...
	const themeName = useSelect( ( select ) =>
		select( coreStore ).getCurrentTheme()
	)?.name?.raw;
	// ...

	return (
		// ...
		<Button>
			{ sprintf(
				__( 'Export "%s" as a Zip File', 'create-block-theme' ),
				themeName
			) }
		</Button>
		// ...
	);
}
```